packages/cli/src/templates/expo.ts

```diff
@@ -0,0 +1,237 @@
+import chalk from "chalk";
+import { Listr } from "listr2";
+import { exec } from "node:child_process";
+import { existsSync } from "node:fs";
+import { readFile, writeFile } from "node:fs/promises";
+import { promisify } from "node:util";
+import { TemplateOptions } from "../commands/create";
+import { Template } from "./template";
+import { modifyJsonFile } from "./utils/modify-json-file";
+
+const execAsync = promisify(exec);
+export interface ExpoContext {
+  options: TemplateOptions;
+}
+
+export const Expo: Template = {
+  name: "expo",
+  description: "A new expo typescript project with gluestack-ui renderer",
+  async handler(options) {
+    await new Listr<ExpoContext>([
+      {
+        title: "Create directory",
+        task: async ({ options: { cwd } }, task) => {
+          task.title += ` ${cwd}`;
+          if (existsSync(cwd)) {
+            throw new Error(`Directory ${cwd} already exists`);
+          }
+          await execAsync(
+            `npx -y create-expo-app ${cwd} -t expo-template-blank-typescript`,
+          );
+        },
+      },
+      {
+        title: "Create Expo project",
+        task: async ({ options: { cwd } }) => {
+          const appJsonFilePath = `${cwd}/app.json`;
+          await modifyJsonFile(appJsonFilePath, (config) => {
+            // @ts-expect-error structure of config file subject to change
```

Comment:
didn't seem worth it to try to resolve this type error (config type of unknown). Wasn't sure the best approach to do that without a lot of extra code. Let me know if you can think of a good solution.

Reply:
I am not a fan of this, although I agree it is not worth typing the config files.
I recommend that you apply my code suggestion in the `modifyJsonFile` method and remove the `@ts-expect-error`.